### PR TITLE
MRG, DOC: adds bem model error solution at the end of fix_bem_in_blender.py tutorial

### PR DESCRIPTION
**Pull request by mkadwani**
#### What does this implement/fix?
Adds Blender tools for fixing bem model errors in the plot_fix_bem_in_blender.py tutorial

### CODE REVIEW

File: tutorials/source-modeling/plot_fix_bem_in_blender.py [hunk 1]
```diff
@@ -135,3 +135,54 @@
...
+# Did you come across these errors after exporting surfaces and running
+# the :func:`mne.make_bem_model` function?
+#
+#
+# * This error below is caused by adding or deleting vertices:
+#
+#   RuntimeError: Cannot decimate to requested ico grade 4. The provided
+#   BEM surface has 20516 triangles, which cannot be isomorphic with a
+#   subdivided icosahedron. Consider manually decimating the surface to a
+#   suitable density and then use ico=None in make_bem_model.
```
Comment (drammock) [inline tutorials/source-modeling/plot_fix_bem_in_blender.py:156]:
```suggestion
# 1. Cannot decimate to requested ico grade
# 
#    This error is caused by having too few or too many vertices. The full
#    error is something like:
#   
#    .. code-block:: console
#    
#        RuntimeError: Cannot decimate to requested ico grade 4. The provided
#        BEM surface has 20516 triangles, which cannot be isomorphic with a
#        subdivided icosahedron. Consider manually decimating the surface to a
#        suitable density and then use ico=None in make_bem_model.
```

File: tutorials/source-modeling/plot_fix_bem_in_blender.py [hunk 2]
```diff
@@ -135,3 +135,54 @@
...
+#
+# * If you have tried to match with the original number of triangles by
+#   decimating vertices, that might have changed the geometry of the
+#   triangular faces:
+#
+#   RuntimeError: Surface inner skull has topological defects: 12 / 20484
+#   vertices have fewer than three neighboring triangles [733, 1014, 2068,
+#   7732, 8435, 8489, 10181, 11120, 11121, 11122, 11304, 11788]
```
Comment (drammock) [inline tutorials/source-modeling/plot_fix_bem_in_blender.py:164]:
```suggestion
# 2. Surface inner skull has topological defects
#
#    This error can occur when trying to match the original number of
#    triangles by removing vertices. The full error looks like:
#   
#    .. code-block:: console
#   
#       RuntimeError: Surface inner skull has topological defects: 12 / 20484
#       vertices have fewer than three neighboring triangles [733, 1014, 2068,
#       7732, 8435, 8489, 10181, 11120, 11121, 11122, 11304, 11788]
```

File: tutorials/source-modeling/plot_fix_bem_in_blender.py [hunk 3]
```diff
@@ -135,3 +135,54 @@
...
+#
+# * Changing the geometry might also result in:
+#
+#   RuntimeError: Surface inner skull is not complete (sum of solid
+#   angles yielded 0.999668, should be 1.)
```
Comment (drammock) [inline tutorials/source-modeling/plot_fix_bem_in_blender.py:169]:
```suggestion
# 3. Surface inner skull is not complete
#
#    This error (like the previous error) reflects a problem with the surface
#    topology (i.e., the expected pattern of vertices/edges/faces is
#    disrupted).
#
#    .. code-block:: console
#   
#       RuntimeError: Surface inner skull is not complete (sum of solid
#       angles yielded 0.999668, should be 1.)
```

File: tutorials/source-modeling/plot_fix_bem_in_blender.py [hunk 4]
```diff
@@ -135,3 +135,54 @@
...
+#
+# * Once the number or order of vertices has been tampered with, it would be
+#   difficult to get back to the original geometry of the surface.
+#
+#   RuntimeError: The source surface has a matching number of
+#   triangles but ordering is wrong
```
Comment (drammock) [inline tutorials/source-modeling/plot_fix_bem_in_blender.py:175]:
```suggestion
# 4. Triangle ordering is wrong
#
#    This error reflects a mismatch between how the surface is represented in
#    memory (the order of the vertex/face definitions) and what is expected by
#    MNE-Python.  The full error is:
#
#    .. code-block:: console
#   
#       RuntimeError: The source surface has a matching number of
#       triangles but ordering is wrong
```

File: tutorials/source-modeling/plot_fix_bem_in_blender.py [hunk 5]
```diff
@@ -135,3 +135,54 @@
...
+#
+# When such errors occur, it is easiest to start afresh. Delete the edited
+# surfaces in Blender. Import the original BEM surface into Blender again, and
+# make changes in it without creation of any new vertices or triangles. For
+# example, select a circle of vertices, then press 'G' to drag them to desired
+# places. Lastly, smooth out the vertices in Blender by right clicking at a
+# group of vertices selected, and selecting Smooth Vertices to get a rounded
+# shape of the surface. These tools do not change the number or order of
+# vertices. This surface when exported out of Blender, and converted to
+# Freesurfer format might not give you the above errors!
```
Comment (drammock) [inline tutorials/source-modeling/plot_fix_bem_in_blender.py:188]:
```suggestion
# For any of these errors, it is usually easiest to start over with the
# unedited BEM surface and try again, making sure to only *move* vertices and
# faces without *adding* or *deleting* any. For example,
# select a circle of vertices, then press :kbd:`G` to drag them to the desired
# location. Smoothing a group of selected vertices in Blender (by
# right-clicking and selecting "Smooth Vertices") can also be helpful.
```

**Review by drammock — COMMENTED** [review]
Thanks for this great contribution!  I've made some suggestions mostly for nicer formatting and smoother prose.  On the "files changed" tab you can add the suggestions to a batch and commit them all at once to this PR (or only some of them, if some suggestions got something wrong).

**Comment by drammock** [comment]
ps: please remove "fixes # 8843" from the description of this PR.  That text gets treated specially by GitHub, and is used for linking a PR to an existing issue that gets resolved by the changes in this PR.  In this case, you've linked the PR to itself, which is a little confusing and might have unexpected consequences when it gets merged in.

**Comment by mkadwani** [comment]
Hi Daniel,

Thank you for the valuable feedback and suggestions. I will make those
changes soon in this week.

Thanks,
Manorama

**Comment by larsoner** [comment]
Lots of `W291 trailing whitespace` errors

https://github.com/mne-tools/mne-python/pull/8843/checks?check_run_id=1907640503#step:7:10

File: tutorials/source-modeling/plot_fix_bem_in_blender.py [hunk 6]
```diff
@@ -135,3 +135,68 @@
...
+# `mne.make_bem_model` function, and the likely causes of those errors.
+#
+#
+# 1. Cannot decimate to requested ico grade
+#
+#    This error is caused by having too few or too many vertices. The full
+#    error is something like:
+#
+#    .. code-block:: console
+#
```
Comment (larsoner) [inline tutorials/source-modeling/plot_fix_bem_in_blender.py:157]:
```suggestion
#
```

File: tutorials/source-modeling/plot_fix_bem_in_blender.py [hunk 7]
```diff
@@ -135,3 +135,68 @@
...
+#       subdivided icosahedron. Consider manually decimating the surface to a
+#       suitable density and then use ico=None in make_bem_model.
+#
+# 2. Surface inner skull has topological defects
+#
+#    This error can occur when trying to match the original number of
+#    triangles by removing vertices. The full error looks like:
+#
+#    .. code-block:: console
+#
```
Comment (larsoner) [inline tutorials/source-modeling/plot_fix_bem_in_blender.py:169]:
```suggestion
#
```

File: tutorials/source-modeling/plot_fix_bem_in_blender.py [hunk 8]
```diff
@@ -135,3 +135,68 @@
...
+#       7732, 8435, 8489, 10181, 11120, 11121, 11122, 11304, 11788]
+#
+# 3. Surface inner skull is not complete
+#
+#    This error (like the previous error) reflects a problem with the surface
+#    topology (i.e., the expected pattern of vertices/edges/faces is
+#    disrupted).
+#
+#    .. code-block:: console
+#
```
Comment (larsoner) [inline tutorials/source-modeling/plot_fix_bem_in_blender.py:181]:
```suggestion
#
```

File: tutorials/source-modeling/plot_fix_bem_in_blender.py [hunk 9]
```diff
@@ -135,3 +135,68 @@
...
+#       angles yielded 0.999668, should be 1.)
+#
+# 4. Triangle ordering is wrong
+#
+#    This error reflects a mismatch between how the surface is represented in
+#    memory (the order of the vertex/face definitions) and what is expected by
+#    MNE-Python.  The full error is:
+#
+#    .. code-block:: console
+#
```
Comment (larsoner) [inline tutorials/source-modeling/plot_fix_bem_in_blender.py:192]:
```suggestion
#
```

**Review by larsoner — COMMENTED** [review]
Found by hitting ctrl-A to select all when viewing the diff on GitHub, though in theory these should be visible in an editor, too. Personally I have my editor remove trailing whitespace from py files automatically when saving.

**Comment by mkadwani** [comment]
Hi, thanks for the help above. I have just one failing check by codecov. I don't understand what that means and what is that 90% target about. Can anyone please explain?

**Review by agramfort — APPROVED** [review]

**Comment by agramfort** [comment]
thx a lot @mkadwani 

codecov can be ignored here.

**Comment by mkadwani** [comment]
Great! Thanks @agramfort

**Comment by drammock** [comment]
Thanks @mkadwani ! FYI, codecov is a tool that checks which lines of our code get used when we run our tests. It helps us improve the tests by showing us where we can increase test coverage. But, it seems to make mistakes often. For this PR it is irrelevant because you only changed a tutorial, not MNE-Python code.

**Comment by mkadwani** [comment]
I see. Thanks for letting me know @drammock !

**Pull request merged**